### PR TITLE
Feature/project layout

### PR DESCRIPTION
**Pull request by Wavesonics**
This is not ready to merge, but I wanted some feedback.

In the `game` sample directory, I moved the `project` contents up one level. So the `kotlin` directory is now a child of the Godot project.

As a child, the `kotlin/bin` dir can now be referenced from the Godot projects, so the GDNative resource can point to the binaries via `res://kotlin/bin/xxx`

There's still kind of a lot of faff in the kotlin dir so I think keeping it some what separate from the Godot files is nice. Plus we can easily tell the Godot project to just ignore the kotlin dir.

### CODE REVIEW

**Comment by chippmann** [comment]
@Wavesonics you are aware of the fact that you also need to change the paths inside the `build.gradle.kts`, and the `samples.gdnlib` right?
Apart from that, i don't have to say much about it. I'm fine with it i guess.
Maybe we want to change the name of the kotlin dir?

**Comment by piiertho** [comment]
I think we can go for something like src/kotlin or script/kotlin.
Also it could be cool to set compiled code in platform directories like bin/macos/libkotlin.dylib or bin/android/arm64/libkotlin.so. So that we do not overwrite other os librairies.

**Comment by Wavesonics** [comment]
@chippmann yeah, I just wanted to get people's input before I started going any further down this road.

Another option: we could stick with something similar to the original setup, where the Kotlin is totally separate, and just have the build script copy the binaries into the godot project. How do the C# projects handle this?

**Comment by Wavesonics** [comment]
@piiertho I thought about src, but the kotlin dir also has an src under it, so: `src/kotlin/src` seemed like it could be confusing. However `script` might be nice. I imagine a truly mad world where script is a common dir used by all languages, and you could mix C# (script/csharp), Rust (script/rust), and Kotlin all in the same project xD

**Comment by piiertho** [comment]
Personnaly I'm fine with this layout.

**Comment by lassem** [comment]
We also have `.gdignore` if there are things we want to hide from the ide.

**Comment by Wavesonics** [comment]
Ok I think the latest CL is in a pretty good spot. Let me know if there is anything else that needs changing.

**Review by piiertho — APPROVED** [review]
This layout seems definitely better than older :)
As it is important changes (it impacts end user experience), I'd like to have @chippmann  and @CedNaru approval too.

**Review by CedNaru — APPROVED** [review]
I'm all for having the kotlin project inside the godot one.
So I'm okay with the changes.

File: samples/coroutines/coroutines.gdnlib [hunk 1]
```diff
@@ -7,12 +7,10 @@ reloadable=true
 
 [entry]
 
-OSX.64="res://libkotlin.dylib"
-Windows.64="res://kotlin.dll"
-X11.64="res://libkotlin.so"
+Windows.64="res://kotlin/build/bin/windows/releaseShared/kotlin.dll"
```
Comment (chippmann) [inline samples/coroutines/coroutines.gdnlib]: On the games sample project we link the debug version and here the release version. I think we shouldlink the debug version both times as they are samples not "finished" games and its more consistent.

**Review by chippmann — DISMISSED** [review]
Apart from that one change, i'm fine with it. Thanks a lot for your work :-)
@piiertho you can merge it once this branch is rebased onto the changes made in the android branch :-)

File: samples/coroutines/coroutines.gdnlib (hunk 1 above)
Comment (piiertho) [reply samples/coroutines/coroutines.gdnlib]: @Wavesonics, can you rebase on master and add requested changes so that we get that merged :)

Comment (piiertho) [reply samples/coroutines/coroutines.gdnlib]: Oh did not see it's coroutine module, it is compiled as release in gradle build script. Can you set it in debug for the moment ? We will add a target parameter to define release or debug.

**Review by piiertho — CHANGES_REQUESTED** [review]
Needs rebase on master.

**Comment by Wavesonics** [comment]
Yeah I can do this tonight

**Comment by Wavesonics** [comment]
@piiertho ok I made both samples build debug, and I rebased on master.

Should be good to go now.

**Review by piiertho — APPROVED** [review]
Thanks !

**Comment by piiertho** [comment]
@chippmann As the changes you requested were done, I consider that ti is ok :)

**Comment by chippmann** [comment]
> @chippmann As the changes you requested were done, I consider that ti is ok :)

@piiertho Sure! Thank you :-)

**Pull request merged**